### ISSUE TITLE
Keras Weighted Cross Entropy (Binary)

### ISSUE THREAD

**Issue by IronLady** [keras-team/keras#12605]
I am trying to implement weighted cross entropy from TF in Keras.
Documentation from TF site : https://www.tensorflow.org/api_docs/python/tf/nn/weighted_cross_entropy_with_logits

This is what I do :

```
import tensorflow as tf
from keras import backend as K

# Create the custom loss function
def weighted_binary_crossentropy(weights):
    def w_binary_crossentropy(y_true, y_pred):
        return K.mean(tf.nn.weighted_cross_entropy_with_logits(
            y_true,
            y_pred,
            weights,
            name=None
        ), axis=-1)
    return w_binary_crossentropy

# Optimizers, Loss and Compile
adam = Adam(lr=0.0001)
weighted_loss = weighted_binary_crossentropy(weights=1)
model.compile(optimizer=adam, loss=weighted_loss, metrics=['accuracy'])
```
Traning started but Loss is not getting updated/stuck. My expectation is if I set the weight to 1, then the result will be the same as standard cross entropy loss.
Did I missed something?

**Comment by hr21** [comment]
Hii @IronLady I guess in such a case tweaking your neural network architechture and changing certain parameters might help or maybe the machine might be expecting a different loss function.

**Comment by opengeostat** [comment]
> I am trying to implement weighted cross entropy from TF in Keras.
> Documentation from TF site : https://www.tensorflow.org/api_docs/python/tf/nn/weighted_cross_entropy_with_logits
> 
> This is what I do :
> 
> ```
> import tensorflow as tf
> from keras import backend as K
> 
> # Create the custom loss function
> def weighted_binary_crossentropy(weights):
>     def w_binary_crossentropy(y_true, y_pred):
>         return K.mean(tf.nn.weighted_cross_entropy_with_logits(
>             y_true,
>             y_pred,
>             weights,
>             name=None
>         ), axis=-1)
>     return w_binary_crossentropy
> 
> # Optimizers, Loss and Compile
> adam = Adam(lr=0.0001)
> weighted_loss = weighted_binary_crossentropy(weights=1)
> model.compile(optimizer=adam, loss=weighted_loss, metrics=['accuracy'])
> ```
> 
> Traning started but Loss is not getting updated/stuck. My expectation is if I set the weight to 1, then the result will be the same as standard cross entropy loss.
> Did I missed something?

Hi @IronLady , did you manage to make this work? I have the same problem, and I need this loss for a very unbalanced semantic segmentation problem.

**Comment by IronLady** [comment]
@opengeostat 

```python
def weighted_binary_crossentropy(weights):
    def w_binary_crossentropy(target, output):
        epsilon_ = tf.convert_to_tensor(K.epsilon(), output.dtype.base_dtype)
        output = clip_ops.clip_by_value(output, epsilon_, 1 - epsilon_)
        output = math_ops.log(output / (1 - output))
    
        return K.mean(tf.nn.weighted_cross_entropy_with_logits(targets=target, logits=output, pos_weight=weights), axis=-1)

    return w_binary_crossentropy```

I can't really remember the correct code. 
Maybe this. Please try.
